core/src/main/java/org/apache/carbondata/core/statusmanager/SegmentUpdateStatusManager.java

```diff
@@ -956,9 +964,16 @@ public UpdateVO getInvalidTimestampRange(String segmentId) {
         long timestamp = CarbonUpdateUtil.getTimeStampAsLong(
             CarbonTablePath.DataFileUtil.getTimeStampFromDeleteDeltaFile(fileName));
 
-        if (block.getBlockName().equalsIgnoreCase(blkName) && (
-            Long.compare(timestamp, deltaStartTimestamp) < 0)) {
-          files.add(eachFile);
+        if (block.getBlockName().equalsIgnoreCase(blkName)) {
+
+          if (getOnlyAbortedFiles) {
+            if (Long.compare(timestamp, deltaEndTimestamp) > 0) {
```

Comment:
Can't you use timestamp > deltaEndTimestamp?

Reply:
yes, but for all the timestamp comparision in code, we were using Long.compare, so i follwed the same , just to make same for all